src/ensembl/src/shared/state/in-app-search/inAppSearchSelectors.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { RootState } from 'src/store';
+import type { AppName, SearchResults } from './inAppSearchSlice';
+
+export const getSearchQuery = (
+  state: RootState,
+  appName: AppName,
+  genomeId: string
+): string => {
+  return state.inAppSearch[appName][genomeId]?.query || '';
```

Comment:
Don't think we need to explicitly specify the return type here and below

Reply:
I think eslint has a [specific rule](https://github.com/typescript-eslint/typescript-eslint/pull/1020) to actually force you to declare the types of the return values of functions when they are exported from a file 😃 We aren't following this rule, and I am happy to remove the types here; but I just wanted to point out that it's more common to err on the side of redundancy 😄

Reply:
✅

Reply:
That's interesting. Is there any benefit in forcing developers to explicitly specify the return type for these exported functions when TypeScript is already doing it for you?

Reply:
I suppose it's double-checking yourself. I.e. if you define the return type of a function explicitly, you guarantee that in the future, even if you refactor your function, its return type will remain the same. Otherwise, when you need to refactor your function in the future, you may inadvertently change its return type, and will not immediately notice that. Typescript will just silently infer a different type from what you may have intended.

This is especially relevant for module exports, where explicitly declared types are guaranteeing a stable public interface.

Reply:
Gotcha. That makes a lot of sense. So should we then consider specifying the types for our module exports in the future?

Reply:
🤷‍♂️

Reply:
@azangru Can we discuss about this in the next code review?